### PR TITLE
Add support to deploy directly to hive clusters.

### PR DESCRIPTION
**Pull request by jewzaam**
https://issues.redhat.com/browse/OSD-4816

- Configuration is updated to support a deploymentMode, Direct or SelectorSyncSet.
- Existing configuration is renamed.
- A template per environment is now created.
- New generate_template.py script creates the templates.
- PR check is updated to verify each of the generated templates.
- A simple configmap in deploy/hive-environment-configmap is created to validate this in all hive envs.
- README.md updated with configuration details.

The old template must remain until app-interface is swapped to use new templates.

### CODE REVIEW

**Comment by jewzaam** [comment]
/hold

Will need to coordinate merge w/ app-interface, deferring to tomorrow.

File: deploy/layered-sre-authorization/config.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+deploymentType: "SelectorSyncSet"
```
Comment (cblecker) [inline deploy/layered-sre-authorization/config.yaml:1]: Shouldn't this be `deploymentMode: "SelectorSyncSet"`?

File: deploy/osd-cluster-admin/config.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+deploymentType: "SelectorSyncSet"
```
Comment (cblecker) [inline deploy/osd-cluster-admin/config.yaml:1]: Shouldn't this be `deploymentMode: "SelectorSyncSet"`?

File: deploy/osd-curated-operatorsources-revert/config.yaml [hunk 3]
```diff
@@ -1,6 +1,8 @@
-matchLabels:
+deploymentType: "SelectorSyncSet"
```
Comment (cblecker) [inline deploy/osd-curated-operatorsources-revert/config.yaml:1]: Shouldn't this be `deploymentMode: "SelectorSyncSet"`?

File: deploy/osd-curated-operatorsources/config.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+deploymentType: "SelectorSyncSet"
```
Comment (cblecker) [inline deploy/osd-curated-operatorsources/config.yaml:1]: Shouldn't this be `deploymentMode: "SelectorSyncSet"`?

File: deploy/sre-prometheus/aws/config.yaml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+deploymentType: "SelectorSyncSet"
```
Comment (cblecker) [inline deploy/sre-prometheus/aws/config.yaml:1]: Shouldn't this be `deploymentMode: "SelectorSyncSet"`?

**Comment by jewzaam** [comment]
@cblecker :palmface:  fixed

**Pull request merged**